src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -2611,169 +2648,468 @@ def list_runs(
-        if len(steps) < self.get_run(run_id).num_steps:
-            return ExecutionStatus.RUNNING
-
-        # Otherwise, return succeeded
-        return ExecutionStatus.COMPLETED
-
-    # ------------------
-    # Pipeline run steps
-    # ------------------
-
-    def get_run_step(self, step_id: UUID) -> StepRunModel:
-        """Get a step by ID.
+    def update_run(self, run: PipelineRunModel) -> PipelineRunModel:
+        """Updates a pipeline run.
 
         Args:
-            step_id: The ID of the step to get.
+            run: The pipeline run to use for the update.
 
         Returns:
-            The step.
+            The updated pipeline run.
 
         Raises:
-            KeyError: if the step doesn't exist.
+            KeyError: if the pipeline run doesn't exist.
         """
         with Session(self.engine) as session:
-            step = session.exec(
-                select(StepRunSchema).where(StepRunSchema.id == step_id)
+            # Check if pipeline run with the given ID exists
+            existing_run = session.exec(
+                select(PipelineRunSchema).where(PipelineRunSchema.id == run.id)
             ).first()
-            if step is None:
+            if existing_run is None:
                 raise KeyError(
-                    f"Unable to get step with ID {step_id}: No step with this "
-                    "ID found."
+                    f"Unable to update pipeline run with ID {run.id}: "
+                    f"No pipeline run with this ID found."
                 )
-            parent_steps = session.exec(
-                select(StepRunSchema)
-                .where(StepRunOrderSchema.child_id == step.id)
-                .where(StepRunOrderSchema.parent_id == StepRunSchema.id)
-            ).all()
-            parent_step_ids = [parent_step.id for parent_step in parent_steps]
-            mlmd_parent_step_ids = [
-                parent_step.mlmd_id for parent_step in parent_steps
-            ]
-            step_model = step.to_model(parent_step_ids, mlmd_parent_step_ids)
-            return step_model
 
-    def get_run_step_outputs(self, step_id: UUID) -> Dict[str, ArtifactModel]:
-        """Get the outputs for a specific step.
+            # Update the pipeline run
+            existing_run.from_update_model(run)
+            session.add(existing_run)
+            session.commit()
+
+            session.refresh(existing_run)
+            return existing_run.to_model()
+
+    def get_run_component_side_effects(
+        self,
+        run_id: UUID,
+        component_id: Optional[UUID] = None,
+    ) -> Dict[str, Any]:
+        """Gets the side effects for a component in a pipeline run.
 
         Args:
-            step_id: The id of the step to get outputs for.
+            run_id: The ID of the pipeline run to get.
+            component_id: The ID of the component to get.
+        """
+        # TODO: raise KeyError if run doesn't exist
+        pass  # TODO
```

Comment:
I'm wondering how mypy isn't complaining about the return type of this method, do we have it disabled for the entire file or something?

Reply:
We do have mypy enabled for the file. I guess mypy doesn't complain about return types if you use `pass`?

Reply:
Mhm I didn't know (and expect) that, but I just checked and you're correct. I guess it would be safer if we raised/asserted something instead, but doesn't matter in this case I think